### ISSUE TITLE
Feature Request: Ability to configure/setup journals within the Ceph Deploy Interface

### ISSUE THREAD

**Issue by bailey-97** [45Drives/cockpit-ceph-deploy#6]
**Feature Request**

**Selecting SSDs to be journals within the Ceph Deploy Interface**

Currently when configuring a Ceph Cluster that utilizes SSDs as journals, after running device aliasing playbook, but prior to running core playbook, you must manually edit each hosts file (/usr/share/ceph-ansible/host_vars/osd1.yml) and move the devices you want to function as journals into the dedicated devices section.

ex. the file when generated appears as:

![picture1](https://user-images.githubusercontent.com/88982700/143462014-0954b25e-3dbe-4d32-8174-a50c031a96bc.PNG)

We must take any devices we want as journals, then move them into the dedicated devices section by editing the file, so it then appears as:

![picture2](https://user-images.githubusercontent.com/88982700/143462052-b65c8242-b369-4392-aa54-15ea3eb96094.PNG)

**Changing the Journal DB size within the Ceph Deploy Interface**

Additionally, if changing the DB size of the parititions on the journals, it requires editing the following value in /usr/share/ceph-ansible/group_vars/all.yml (line 261):

**block_db_size: 32212254720 # block db size in bytes for the ceph-volume lvm batch. -1 means use the default of 'as big as possible'.**

**Request**

If the functionality to select both devices that should be journals within the UI without needing to edit each file by hand in the terminal, as well as the ability to edit the journal partition sizes within the UI without editing group_vars/all.yml within the terminal were added, it would be much appreciated!

If you require any further information please let me know.

**Comment by markdhooper** [comment]
One of these is easier than the other.
I can add a block_db_size option to all.yml pretty easily. Is `32212254720` a good default? or should it be `-1`? (going with defaults from ceph-deploy-45d for what they are currently.)

The dedicated devices section is a bit more involved, but I will brainstorm with you/Brett to see how we want to put it in, plus any other parameters that need to be set when gearing that up. (`journal_size`) for instance.

**Comment by bkelly16** [comment]
lets use -1 as the default, for "block_db_size" its best to use as much space of the SSD as possible.

**Comment by markdhooper** [comment]
OK, so plan is to set up `block_db_size` as its own option first. That is easy and I can get that configurable in a quick patch.

For the dedicated device feature, I need to make some modifications to `generate-osd-vars.sh` with the ability to omit devices using some kind of arguments, as the dedicated devices can't also be included in the list of devices.

Since the `device-alias.yml` playbook runs that script, I'll need to modify `ceph-ansible-45d` as well. I was thinking of setting a variable in `all.yml`. (a checkbox in ceph deploy) like `use_dedicated_db_devices` or whatever. You can toggle that on/off in the ansible configuration and provide device path(s) for each osd node that you want to use as the dedicated device.

I'll force you to use `/dev/disk/by-vdev/` at the start of each new path, then you just type in the device alias you want to use on a given osd node. 

That info will make its way into each osd node's inventory file, and the `device-alias.yml` playbook will just call `generate-osd-vars.sh` with arguments for which devices to exclude from the device list. 

So that is a bit more involved, and will have to get its own release after.

Does that sound like a plan?

**Comment by bailey-97** [comment]
that sounds very good to me, thank you sir

**Comment by markdhooper** [comment]
I just added the block_db_size option, with the default set to -1.

**Comment by markdhooper** [comment]
Just added the required updates to 45Drives-tools to make this work.
You can now give generate-osd-vars.sh a -d option, which will allow you to provide it a comma separated list of dedicated journal device paths to exclude from the output.

BEFORE:
```
[root@rockinator1 tools]# ./generate-osd-vars.sh
---
osd_auto_discovery: false

devices:
  - /dev/disk/by-vdev/1-6
  - /dev/disk/by-vdev/1-8
  - /dev/disk/by-vdev/1-10
  - /dev/disk/by-vdev/1-13
  - /dev/disk/by-vdev/1-14
  - /dev/disk/by-vdev/1-15
```

AFTER:
```
[root@rockinator1 tools]# ./generate-osd-vars.sh -d /dev/disk/by-vdev/1-14,/dev/disk/by-vdev/1-15
---
osd_auto_discovery: false

devices:
  - /dev/disk/by-vdev/1-6
  - /dev/disk/by-vdev/1-8
  - /dev/disk/by-vdev/1-10
  - /dev/disk/by-vdev/1-13

```

Now I'm going to look at adding the UI elements for enabling the journaling/setting dedicated devices on the osds.

**Comment by markdhooper** [comment]
I just released [cockpit-ceph-deploy v1.0.2](https://github.com/45Drives/cockpit-ceph-deploy/releases/tag/v1.0.2)

Within the osd options there is now an option "Use Dedicated Device DB". You can turn it on and specify the devices on each osd to use as your dedicated journaling devices by filling in the path eg `/dev/disk/by-vdev/1-15` for each osd.

![updated options](https://user-images.githubusercontent.com/50240103/145838579-6a997b9b-dcdd-4c71-bb7e-3992cf504949.png)
